### DISCUSSION TITLE
Problem with parallel routes and server actions

### DISCUSSION THREAD

**Discussion by Altro31** [vercel/next.js#58324] [App Router] [answered]
I have a problem with parallel routes. I'm using its with server actions. So, if I use:
- redirect: Not works.
- revalidatePath or revalidateTag: the form go into an infinity submit status.

Those problems only ocurs when a parallel route is active. Elsewhere, all works well.

**Comment by Quatton** [comment]
Hi. I got the same issue as well. I use `useFormState` and `useFormStatus`. `useFormState` never returns and `useFormStatus` is pending forever.

**Reply by Viozhu** [reply]
Any solution?

**Reply by renekahr** [reply]
Same issue here. I'm on nextjs 15.1.6. 
I use `useActionState` which never returns when there is an `active` parallel route. So `pending` is `true` forever.

**Comment by Altro31** [comment, marked as answer]
**v14.0.4** resolves the problem 🥳🥳🥳. But, i still have it with Turbopack enabled

**Comment by Altro31** [comment]
This issue was solved in the 14.1
________________________________
De: Jorge Ignacio Garay ***@***.***>
Enviado: viernes, 2 de febrero de 2024 16:08
Para: vercel/next.js ***@***.***>
Cc: Altro ***@***.***>; Manual ***@***.***>
Asunto: Re: [vercel/next.js] Problem with parallel routes and server actions (Discussion #58324)


Any solution?

**Reply by Viozhu** [reply]
But the problem still with useFormState never returns and useFormStatus is pending forever.

**Reply by Quatton** [reply]
For now it could be fixed with [...catchAll] routes because default.tsx doesn't work as expected. 

Simply export default the default page to [...catchAll]/page.tsx
